src/dc_tools.rs

```diff
@@ -386,7 +386,7 @@ pub(crate) fn dc_extract_grpid_from_rfc724_mid(mid: &str) -> Option<&str> {
         if let Some(grpid_len) = mid_without_offset.find('.') {
             /* strict length comparison, the 'Gr.' magic is weak enough */
             if grpid_len == 11 || grpid_len == 16 {
-                return Some(mid_without_offset.get(0..grpid_len).unwrap());
+                return Some(mid_without_offset.get(0..grpid_len).unwrap_or_default());
```

Comment:
fairly sure all the bounds are already checked in the various if statements above and any remaining mistake is a programming error.